helpers/telemetry.py

```diff
@@ -32,114 +30,56 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-class MetricContext:
+@sync_to_async
+def log_simple_metric_async(name: str, value: float):
+    log_simple_metric(name, value)
+
+
+def log_simple_metric(name: str, value: float):
     """
-    Timeseries metrics can be tagged with context (repo, commit, owner) and this
-    class holds onto that context. It exposes a way to log timeseries metrics
-    with context appended.
+    `log_simple_metric()` will get metadata values from the log context
+    and then log a simple metric with the pass-in name/value to Postgres.
+    This function is synchronous/blocking.
+    """
+
+    # Timezone-aware timestamp in UTC
+    timestamp = django.utils.timezone.now()
 
-    Create it with whatever context you've got (e.g. IDs that were passed in as
-    task arguments) and it will attempt to fetch the rest before logging for the
-    first time.
+    log_context = get_log_context()
 
-    `log_simple_metric()` will call `populate()` if it hasn't been called before
-    and then log a simple metric with the pass-in name/value to both Postgres
-    and Timescale. This function is synchronous/blocking.
+    try:
+        PgSimpleMetric.objects.create(
+            timestamp=timestamp,
+            name=name,
+            value=value,
+            repo_id=log_context.repo_id,
+            owner_id=log_context.owner_id,
+            commit_id=log_context.commit_id,
+        )
+    except Exception:
+        log.exception("Failed to create telemetry_simple record")
 
+
+@fire_and_forget
+async def attempt_log_simple_metric(name: str, value: float):
+    """
     `attempt_log_simple_metric()` is the @fire_and_forget version of
-    `log_simple_metric()`, meaning it will throw populating/logging on the
+    `log_simple_metric()`, meaning it will throw logging on the
     asyncio queue and then immediately return without needing to be awaited. Can
```

Comment:
as this is now using a `ContextVar`, is that guaranteed to be more consistent than a global which might change underneath?

Reply:
y'know, good question. i think so; as i understand it the context will inherit a clone of the parent context so the parent context is free to change